### DISCUSSION TITLE
Repeating particles starting from second time step

### DISCUSSION THREAD

**Discussion by JonathanC33** [Parcels-code/Parcels#1154] [General]
Hi, I'm trying to create a repeating particle output every hour, with position and time recorded every hour. The particles are initially defined in a gridded formation but the repeats of the particles are initially placed at the second time step of the first particles released. Is there a way to define at which time step the repeating particle positions are take from?

#Fieldset using FieldSet.from_netcdf using the filenames, variables, and dimensions
fieldset_ocean = FieldSet.from_netcdf(
	filenames, 
	variables, 
	dimensions,
    indices=indices,
    deferred_load=True,
	allow_time_extrapolation=True) 

#Run through each of the ToEs, setting up particle grids and repeating every 3 hours
#Run for 2 days total to cover the entire time of interest
#Mask out land for the lons and lats used later for particle locations 
#Every 5th space, approx. 600m spacing 
lons = (lon*land_mask)[::5,::5]
lats = (lat*land_mask)[::5,::5]
#Release new particles every x hour(s) 
repeatdt = timedelta(hours=1)

for i in ToI.keys():
    for ii in ToI[i].keys():
        
        #Take the first day of each ToI
        iii = ToI[i][ii][0]
        
        
        #Define the ToE
        OP_start = np.datetime64(iii[:4]+'-'+iii[4:6]+'-'+iii[6:])
        #Define Particle Locations, and time according to ToE
        pset = ParticleSet.from_list(fieldset=fieldset_ocean, pclass=JITParticle,
        lon=lons.flatten()[~np.isnan(lons.flatten())],
        lat=lats.flatten()[~np.isnan(lats.flatten())],
        time=OP_start,
        repeatdt=repeatdt)
        
        #Don't save the depth of the particles (always 0)
        pset.set_variable_write_status('depth', False)
        
        def DeleteParticle(particle, fieldset, time):
            particle.delete()
            
        #Run the OP output for 2 days
        output_file = pset.ParticleFile(name='/gpfs/fs4/dfo/dpnm/dfo_dpnm/joc000/OceanParcelsOutput/ToI/'+\
                                        i+'/ToI-GriddedRelease_'+\
                                        i+'_'+iii+\
                                        '_100m-res_1hr-repeat.nc',
                                        outputdt=timedelta(minutes=60))
            
        kernels = pset.Kernel(AdvectionRK4) #+ SampleAge
        pset.execute(kernels,
        runtime=timedelta(hours=24*2),
        dt=timedelta(minutes=1),
        recovery={ErrorCode.ErrorOutOfBounds: DeleteParticle},
        output_file=output_file)
        output_file.export()

**Comment by JonathanC33** [comment]
I've also checked the pset.repeatlon and pset.repeatlat variables. At the first time step, they are in the correct location, but after that they shift to the second time step particle locations and that becomes the repeat particle locations for the remainder of the execution.

**Comment by erikvansebille** [comment]
Hi @JonathanC33. If you want more control of particle start times and locations, I'd suggest to use the 'delayed release' functionality (i.e. also provide a `time=` argument to the ParticleSet creation). See also the 'Assigning each particle its own `time` section of the [delay start tutorial](https://nbviewer.org/github/OceanParcels/parcels/blob/master/parcels/examples/tutorial_delaystart.ipynb#Assigning-each-particle-its-own-time).
